### ISSUE TITLE
Report "images don't load, see log"

### ISSUE THREAD

**Issue by K1rakishou** [Adamantcheese/Kuroba#1094]
UUID: 8ac6cb1c89b4edea55163000e71a908f
Build flavor: stable
Version name: v4.13.6
OS info: Android 9, sdk version: 28
Report description: my internet is fine. sometimes KUROBA gets in a bad state where none of the images load and I gotta manually clear the cache and restart my phone to fix it. swiping through the gallery shows toast errors on every image
Logs (truncated):
```
E/Kuroba | CacheHandler: Error while trying to get or create cache file
E/Kuroba | CacheHandler: java.io.IOException: No such file or directory
E/Kuroba | CacheHandler: 	at java.io.UnixFileSystem.createFileExclusively0(Native Method)
E/Kuroba | CacheHandler: 	at java.io.UnixFileSystem.createFileExclusively(UnixFileSystem.java:281)
E/Kuroba | CacheHandler: 	at java.io.File.createNewFile(File.java:1008)
E/Kuroba | CacheHandler: 	at com.github.k1rakishou.fsaf.manager.RawFileManager.create(RawFileManager.kt:68)
E/Kuroba | CacheHandler: 	at com.github.k1rakishou.fsaf.manager.RawFileManager.create(RawFileManager.kt:14)
E/Kuroba | CacheHandler: 	at com.github.k1rakishou.fsaf.FileManager.create(FileManager.kt:358)
E/Kuroba | CacheHandler: 	at com.github.k1rakishou.fsaf.FileManager.create(FileManager.kt:330)
E/Kuroba | CacheHandler: 	at com.github.adamantcheese.chan.core.cache.CacheHandler.getOrCreateCacheFile(CacheHandler.kt:120)
E/Kuroba | CacheHandler: 	at com.github.adamantcheese.chan.core.cache.FileCacheV2.enqueueDownloadFileRequest(FileCacheV2.kt:249)
E/Kuroba | CacheHandler: 	at com.github.adamantcheese.chan.core.cache.FileCacheV2.enqueueChunkedDownloadFileRequest(FileCacheV2.kt:229)
E/Kuroba | CacheHandler: 	at com.github.adamantcheese.chan.ui.view.PostImageThumbnailView.setPostImage(PostImageThumbnailView.java:95)
E/Kuroba | CacheHandler: 	at com.github.adamantcheese.chan.ui.cell.PostCell.buildThumbnails(PostCell.java:653)
E/Kuroba | CacheHandler: 	at com.github.adamantcheese.chan.ui.cell.PostCell.bindPost(PostCell.java:353)
E/Kuroba | CacheHandler: 	at com.github.adamantcheese.chan.ui.cell.PostCell.setPost(PostCell.java:298)
E/Kuroba | CacheHandler: 	at com.github.adamantcheese.chan.ui.adapter.PostAdapter.onBindViewHolder(PostAdapter.java:140)
E/Kuroba | CacheHandler: 	at androidx.recyclerview.widget.RecyclerView$Adapter.onBindViewHolder(RecyclerView.java:7065)
E/Kuroba | CacheHandler: 	at androidx.recyclerview.widget.RecyclerView$Adapter.bindViewHolder(RecyclerView.java:7107)
E/Kuroba | CacheHandler: 	at androidx.recyclerview.widget.RecyclerView$Recycler.tryBindViewHolderByDeadline(RecyclerView.java:6012)
E/Kuroba | CacheHandler: 	at androidx.recyclerview.widget.RecyclerView$Recycler.tryGetViewHolderForPositionByDeadline(RecyclerView.java:6279)
E/Kuroba | CacheHandler: 	at androidx.recyclerview.widget.GapWorker.prefetchPositionWithDeadline(GapWorker.java:288)
E/Kuroba | CacheHandler: 	at androidx.recyclerview.widget.GapWorker.flushTaskWithDeadline(GapWorker.java:345)
E/Kuroba | CacheHandler: 	at androidx.recyclerview.widget.GapWorker.flushTasksWithDeadline(GapWorker.java:361)
E/Kuroba | CacheHandler: 	at androidx.recyclerview.widget.GapWorker.prefetch(GapWorker.java:368)
E/Kuroba | CacheHandler: 	at androidx.recyclerview.widget.GapWorker.run(GapWorker.java:399)
E/Kuroba | CacheHandler: 	at android.os.Handler.handleCallback(Handler.java:873)
E/Kuroba | CacheHandler: 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/Kuroba | CacheHandler: 	at android.os.Looper.loop(Looper.java:193)
E/Kuroba | CacheHandler: 	at android.app.ActivityThread.main(ActivityThread.java:6680)
E/Kuroba | CacheHandler: 	at java.lang.reflect.Method.invoke(Native Method)
E/Kuroba | CacheHandler: 	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:493)
E/Kuroba | CacheHandler: 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:858)
```

**Comment by K1rakishou** [comment]
Hard to say what could cause this, I would start with adding more logs.

**Comment by Adamantcheese** [comment]
It's probably best that you handle IOException in FSAF for any File.createNewFile() calls and return null instead of letting the exception pass through; all of the necessary logging for what could be going wrong is already in place as far as I can see.

**Comment by K1rakishou** [comment]
Well I decided to let the caller handle all the exceptions precisely because of crashes like this one where it's not obvious what's going on.

>all of the necessary logging for what could be going wrong is already in place as far as I can see.

I mean like logging whether the directory actually exists or not when this crash occurs. By the way you should probably update FSAF to the latest version (v42) because I did couple of fixes to it.

**Comment by Adamantcheese** [comment]
Closed with closure of #1071.
